### PR TITLE
aws-sam-cli: fix missing `cfn-lint` dep

### LINKED ISSUE #137558: AWS SAM CLI missing dependency since 1.92.0
### `brew gist-logs <formula>` link OR `brew config` AND `brew doctor` output

```shell
$ brew config
HOMEBREW_VERSION: 4.1.1-31-g471e733
ORIGIN: https://github.com/Homebrew/brew
HEAD: 471e733641284b27b7cb98e7329f8efb70fe1dbf
Last commit: 2 hours ago
Core tap origin: https://github.com/Homebrew/homebrew-core
Core tap HEAD: a03b5b21d11377f90e0e1529fe9581185e95c1e5
Core tap last commit: 7 days ago
Core tap branch: master
Core tap JSON: 25 Jul 17:07 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 10
Homebrew Ruby: 2.6.10 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: 10-core 64-bit arm_firestorm_icestorm
Clang: 14.0.0 build 1400
Git: 2.37.1 => /Library/Developer/CommandLineTools/usr/bin/git
Curl: 7.87.0 => /usr/bin/curl
macOS: 12.6.6-arm64
CLT: 14.2.0.0.1.1668646533
Xcode: N/A
Rosetta 2: false
```

```
$ brew doctor
Your system is ready to brew.
```

### Verification

- [X] My "`brew doctor` output" says `Your system is ready to brew.` and am still able to reproduce my issue.
- [X] I ran `brew update` and am still able to reproduce my issue.
- [X] I have resolved all warnings from `brew doctor` and that did not fix my problem.
- [X] I searched for recent similar issues at https://github.com/Homebrew/homebrew-core/issues?q=is%3Aissue and found no duplicates.

### What were you trying to do (and why)?

Issue on the AWS SAM CLI repository: https://github.com/aws/aws-sam-cli/issues/5608

Trying to run `sam validate --lint` to validate a Cloudformation template after installing AWS SAM CLI v1.93.0 fails due to a missing dependency.

### What happened (include all command output)?

Running `sam validate --lint` raises an exception due to a missing package:
```
~/sam-app ⌚ 10:10:22
$ brew install aws-sam-cli
==> Fetching aws-sam-cli
==> Downloading https://ghcr.io/v2/homebrew/core/aws-sam-cli/manifests/1.93.0
Already downloaded: /Users/lucashuy/Library/Caches/Homebrew/downloads/b098f5a720caa22ad99ab0c80661fab2f598b900d6f37eb52a6549faacd3da50--aws-sam-cli-1.93.0.bottle_manifest.json
==> Downloading https://ghcr.io/v2/homebrew/core/aws-sam-cli/blobs/sha256:6648476abcdcd49c8067fa1718
############################################################################################# 100.0%
==> Pouring aws-sam-cli--1.93.0.arm64_monterey.bottle.tar.gz
🍺  /opt/homebrew/Cellar/aws-sam-cli/1.93.0: 7,986 files, 158.7MB
==> Running `brew cleanup aws-sam-cli`...
Disable this behaviour by setting HOMEBREW_NO_INSTALL_CLEANUP.
Hide these hints with HOMEBREW_NO_ENV_HINTS (see `man brew`).

~/sam-app ⌚ 10:10:54
$ sam validate --lint

Error: No module named 'cfnlint'
Traceback:
  File "/opt/homebrew/Cellar/aws-sam-cli/1.93.0/libexec/lib/python3.11/site-packages/click/core.py", line 1055, in main
    rv = self.invoke(ctx)
         ^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/aws-sam-cli/1.93.0/libexec/lib/python3.11/site-packages/click/core.py", line 1657, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/aws-sam-cli/1.93.0/libexec/lib/python3.11/site-packages/click/core.py", line 1404, in invoke
    return ctx.invoke(self.callback, **ctx.params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/aws-sam-cli/1.93.0/libexec/lib/python3.11/site-packages/click/core.py", line 760, in invoke
    return __callback(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/aws-sam-cli/1.93.0/libexec/lib/python3.11/site-packages/click/decorators.py", line 84, in new_func
    return ctx.invoke(f, obj, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/aws-sam-cli/1.93.0/libexec/lib/python3.11/site-packages/click/core.py", line 760, in invoke
    return __callback(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/aws-sam-cli/1.93.0/libexec/lib/python3.11/site-packages/samcli/lib/telemetry/metric.py", line 184, in wrapped
    raise exception  # pylint: disable=raising-bad-type
    ^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/aws-sam-cli/1.93.0/libexec/lib/python3.11/site-packages/samcli/lib/telemetry/metric.py", line 149, in wrapped
    return_value = func(*args, **kwargs)
                   ^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/aws-sam-cli/1.93.0/libexec/lib/python3.11/site-packages/samcli/lib/utils/version_checker.py", line 42, in wrapped
    actual_result = func(*args, **kwargs)
                    ^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/aws-sam-cli/1.93.0/libexec/lib/python3.11/site-packages/samcli/cli/main.py", line 95, in wrapper
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/aws-sam-cli/1.93.0/libexec/lib/python3.11/site-packages/samcli/commands/_utils/cdk_support_decorators.py", line 39, in wrapped
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/aws-sam-cli/1.93.0/libexec/lib/python3.11/site-packages/samcli/commands/_utils/command_exception_handler.py", line 88, in wrapper_command_exception_handler
    raise ex
  File "/opt/homebrew/Cellar/aws-sam-cli/1.93.0/libexec/lib/python3.11/site-packages/samcli/commands/_utils/command_exception_handler.py", line 68, in wrapper_command_exception_handler
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/aws-sam-cli/1.93.0/libexec/lib/python3.11/site-packages/samcli/commands/validate/validate.py", line 58, in cli
    do_cli(ctx, template_file, lint)  # pragma: no cover
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/aws-sam-cli/1.93.0/libexec/lib/python3.11/site-packages/samcli/commands/validate/validate.py", line 73, in do_cli
    _lint(ctx, template)
  File "/opt/homebrew/Cellar/aws-sam-cli/1.93.0/libexec/lib/python3.11/site-packages/samcli/commands/validate/validate.py", line 146, in _lint
    import cfnlint.core  # type: ignore
    ^^^^^^^^^^^^^^^^^^^

An unexpected error was encountered while executing "sam validate".
Search for an existing issue:
https://github.com/aws/aws-sam-cli/issues?q=is%3Aissue+is%3Aopen+Bug%3A%20sam%20validate%20-%20ModuleNotFoundError
Or create a bug report:
https://github.com/aws/aws-sam-cli/issues/new?template=Bug_report.md&title=Bug%3A%20sam%20validate%20-%20ModuleNotFoundError
```

### What did you expect to happen?

I expect that the validate command will not print a stack trace and for the `cfnlint` library to be resolved.

Reverting https://github.com/Homebrew/homebrew-core/commit/a6e04d4e6d3ba319680b1d5ad9a958ff86410e33 that adds `cfn-lint` as a brew dependency and removes the `cfnlint` resource should resolve the missing Python package.

### Step-by-step reproduction instructions (by running `brew` commands)

```shell
1. `brew install aws-sam-cli`
1. `sam init --name test --runtime python3.8 --dependency-manager pip --app-template hello-world`
1. `cd test`
1. `sam validate --lint`
```

### PR DESCRIPTION
**Pull request by branchv**
- [ ] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [ ] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [ ] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

---

Fixes #137558

Would need a `.pth` file for python to find `cfn-lint`, but I don't think we gain much with the added complexity since it also

- Ties these two formulae to a single python version, making our python migration more difficult
- Still duplicates all of `cfn-lint`'s pip resources, unless we also exclude those (which is a manual maintence burden that I'd prefer to leave to tooling like `update-python-resources`)

### CODE REVIEW

File: Formula/aws-sam-cli.rb [hunk 1]
```diff
@@ -57,20 +56,25 @@ class AwsSamCli < Formula
...
 
   resource "certifi" do
     url "https://files.pythonhosted.org/packages/98/98/c2ff18671db109c9f10ed27f5ef610ae05b73bd876664139cf95bd1429aa/certifi-2023.7.22.tar.gz"
     sha256 "539cc1d13202e33ca466e88b2807e29f4c13049d6d87031a3c110744495cb082"
   end
 
+  resource "cfn-lint" do
+    url "https://files.pythonhosted.org/packages/92/2c/e6ac940326e00872bd00f5ebb3c925ca1a9ab064917aa3bcf8e668e0ea83/cfn-lint-0.78.2.tar.gz"
+    sha256 "363787653b71cd5dc4e2de30aaab835afd48b64e2257fe1b4b78efac01635e8c"
+  end
```
Comment (chenrui333) [inline Formula/aws-sam-cli.rb:76]: so it can be covered with pth?

Comment (chenrui333) [reply Formula/aws-sam-cli.rb:76]: oh, i see you already made in the comment.

**Review by chenrui333 — APPROVED** [review]

**Comment by chenrui333** [comment]
> * Ties these two formulae to a single python version, making our python migration more difficult

I find this might be the case with the #137226 idea, but it would also make rust upgrade challenging as well. 

Compared with how often we upgraded python version vs how often we do cfn-lint/aws-sam-cli updates, I still feel like pth work would be valuable.

**Comment by branchvincent** [comment]
Oh was the idea to remove `rust` here? That tradeoff makes sense, but we would also need to remove `cryptography` right

**Comment by chenrui333** [comment]
> Oh was the idea to remove `rust` here? That tradeoff makes sense, but we would also need to remove `cryptography` right

yeah, removing all rust build dependencies for python formulae (but still I'd like to revisit the idea of link more python formulae, not pressing as rust build dependency stuff though)

**Pull request merged**